src/Features/Core/Portable/Wrapping/Edit.cs

```diff
@@ -63,9 +65,27 @@ public static Edit DeleteBetween(SyntaxNodeOrToken left, SyntaxNodeOrToken right
             SyntaxNodeOrToken left, SyntaxTriviaList leftTrailingTrivia,
             SyntaxTriviaList rightLeadingTrivia, SyntaxNodeOrToken right)
         {
-            var leftLastToken = left.IsToken ? left.AsToken() : left.AsNode().GetLastToken();
-            var rightFirstToken = right.IsToken ? right.AsToken() : right.AsNode().GetFirstToken();
+            var leftLastToken = left.IsToken ? left.AsToken() : left.AsNode()!.GetLastToken();
+            var rightFirstToken = right.IsToken ? right.AsToken() : right.AsNode()!.GetFirstToken();
             return new Edit(leftLastToken, leftTrailingTrivia, rightFirstToken, rightLeadingTrivia);
         }
+
+        private sealed class InvalidEditException : Exception
+        {
+            // Used for analyzing dumps
+#pragma warning disable IDE0052 // Remove unread private members
+            private readonly SyntaxTree? _tree;
+            private readonly SyntaxToken _left;
+            private readonly SyntaxToken _right;
+#pragma warning restore IDE0052 // Remove unread private members
+
+            public InvalidEditException(SyntaxToken left, SyntaxToken right)
+                : base($"Left token: '{left}' had an end past the start of right token: '{right}'")
```

Comment:
if we can't keep trees, can we actually keep tokens in the message?

Reply:
@jasonmalinowski this should pii information right?